src/iaaft.jl

```diff
@@ -1,3 +1,110 @@
+"""   
+    IAAFT([x,])
+
+An iteratively adjusted amplitude-adjusted-fourier-transform surrogate[^SchreiberSchmitz1996].
+
+If the timeseries `x` is provided, fourier transforms are planned, enabling more efficient
+use of the same method for many surrogates of a signal with same length and eltype as `x`.
+
+## References 
+
+[^SchreiberSchmitz1996]: T. Schreiber; A. Schmitz (1996). "Improved Surrogate Data for Nonlinearity 
+    Tests". Phys. Rev. Lett. 77 (4): 635–638. doi:10.1103/PhysRevLett.77.635.
+    PMID 10062864. [https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.77.635](https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.77.635)
+"""
+struct IAAFT{F, I} <: Surrogate
+    forward::F
+    inverse::I
+    n_maxiter::Int
+    tol::Real
+    n_windows::Int
```

Comment:
Maybe its worth considering some more consice names for these variables?

Reply:
Hm, yes.

- `n_maxiter` is the number of maximum allowed iterations before the adjustment procedure terminates (the procedure is typically terminated much earlier if sufficient convergence between the periodogram of the original signal and the surrogate's periodogram).
- `tol` gives this tolerance.
- To determine convergence at the i-th, I coarse grain the periodograms for both the original signal and the i-th adjusted surrogate.  `n_windows` is the number of frequency bins over which the power is averaged.

Any good suggestions for what to call these? 
- `n_maxiter`->`maxiters`? 
- `n_windows` -> `nfrequencybins`/`nfreqbins`/`nbins`? 
- `tol` -> `tolerance`/`conv_tol`/`convergence_tolerance` (or keep  as is?)

Reply:
I'd say: `n_maxiter -> M, n_windows -> W, tol -> tol` :D I tend to not have explanatory names for keywords that explained anyway in the docstring. It is true that clarity is to be preferred, but for a user that already now what they are, I think most would prefer the concise syntax.